tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,610 @@
+                   @JsonProperty("totalConsumerConnections") long totalConsumerConnections,
+                   @JsonProperty("totalConsumerFailedConnections") long totalConsumerFailedConnections,
+                   @JsonProperty("totalMetadataConnections") long totalMetadataConnections,
+                   @JsonProperty("totalMetadataFailedConnections") long totalMetadataFailedConnections,
+                   @JsonProperty("totalAborts") long totalAborts,
+                   @JsonProperty("updatedMs") long updatedMs) {
+            this.totalProducerConnections = totalProducerConnections;
+            this.totalProducerFailedConnections = totalProducerFailedConnections;
+            this.totalConsumerConnections = totalConsumerConnections;
+            this.totalConsumerFailedConnections = totalConsumerFailedConnections;
+            this.totalMetadataConnections = totalMetadataConnections;
+            this.totalMetadataFailedConnections = totalMetadataFailedConnections;
+            this.totalAborts = totalAborts;
+            this.updatedMs = updatedMs;
+        }
+
+        @JsonProperty
+        public long totalProducerConnections() {
+            return totalProducerConnections;
+        }
+
+        @JsonProperty
+        public long totalProducerFailedConnections() {
+            return totalProducerFailedConnections;
+        }
+
+        @JsonProperty
+        public long totalConsumerConnections() {
+            return totalConsumerConnections;
+        }
+
+        @JsonProperty
+        public long totalConsumerFailedConnections() {
+            return totalConsumerFailedConnections;
+        }
+
+        @JsonProperty
+        public long totalMetadataConnections() {
+            return totalMetadataConnections;
+        }
+
+        @JsonProperty
+        public long totalMetadataFailedConnections() {
+            return totalMetadataFailedConnections;
+        }
+
+        @JsonProperty
+        public long totalAborts() {
+            return totalAborts;
+        }
+
+        @JsonProperty
+        public long updatedMs() {
+            return updatedMs;
+        }
+    }
+
+    @Override
+    public void stop(Platform platform) throws Exception {
+        if (!running.compareAndSet(true, false)) {
+            throw new IllegalStateException("SustainedConnectionWorker is not running.");
+        }
+        log.info("{}: Deactivating SustainedConnectionWorker.", id);
+
+        // Shut down the periodic status updater and perform a final update on the
+        // statistics.  We want to do this first, before deactivating any threads.
+        // Otherwise, if some threads take a while to terminate, this could lead
+        // to a misleading rate getting reported.
+        this.statusUpdaterFuture.cancel(false);
+        this.statusUpdaterExecutor.shutdown();
+        this.statusUpdaterExecutor.awaitTermination(1, TimeUnit.HOURS);
+        this.statusUpdaterExecutor = null;
+        new StatusUpdater().run();
```

Comment:
This is I guess to overwrite the status to an empty value? Do we need to clean the state?

Reply:
TBH, I copied this from one of the other workers.  I don't feel the need to clean the state though.

Reply:
This won't update the status to an empty value, it just runs the status updater again immediately rather than waiting for the next cycle..  The class itself is stateless.

Reply:
👍